### PR TITLE
gh-101101: Unstable C API tier (PEP 689)

### PR DESCRIPTION
**Pull request by encukou**
This is the `cpython` part of PEP-689 initial implementation.

* Issue: gh-101101

### CODE REVIEW

**Comment by encukou** [comment]
Oh, I forgot blurb. Will add one a bit later.

File: Doc/c-api/stable.rst [hunk 1]
```diff
@@ -18,8 +18,38 @@ way; see :ref:`stable-abi-platform` below).
...
+.. _unstable-c-api:
+
+Unstable C API
+==============
+
+.. index:: single: PyUnstable
+
+Any API named with the ``PyUnstable`` prefix exposes CPython implementation
+details, and may change in every minor release (e.g. from 3.9 to 3.10) without
+any deprecation warnings.
```
Comment (arhadthedev) [inline Doc/c-api/stable.rst:46]: Should `warning` be here instead of `warnings`?

Comment (encukou) [reply Doc/c-api/stable.rst:46]: I think it's fine either way.

**Comment by encukou** [comment]
Oops, I forgot to push the blurb commit. Sorry for the delay.

**Pull request merged**